accessibilityTest/AccessibilityCategoryFacet.ts

```diff
@@ -0,0 +1,80 @@
+import * as axe from 'axe-core';
+import { $$, Component, CategoryFacet, get, Dom } from 'coveo-search-ui';
+import { afterDeferredQuerySuccess, getFacetColumn, getRoot, inDesktopMode, resetMode, waitUntilSelectorIsPresent } from './Testing';
+import { ContrastChecker } from './ContrastChecker';
+
+export const AccessibilityCategoryFacet = () => {
+  describe('CategoryFacet', () => {
+    const getFacetElement = () => {
+      return $$('div', { className: Component.computeCssClassName(CategoryFacet), 'data-field': '@objecttype' });
+    };
+
+    beforeEach(() => {
+      inDesktopMode();
+    });
+
+    afterEach(() => {
+      resetMode();
+    });
+
+    it('should be accessible', async done => {
+      getFacetColumn().appendChild(getFacetElement().el);
+      await afterDeferredQuerySuccess();
+      const axeResults = await axe.run(getRoot());
+      expect(axeResults).toBeAccessible();
+      done();
+    });
+
+    describe('with facet element', () => {
+      let facetElement: Dom;
+
+      function getMagnifierSVG() {
+        return facetElement.el.querySelector<HTMLElement>('.coveo-facet-search-magnifier-svg');
+      }
+
+      function getSearchButton() {
+        return facetElement.el.querySelector<HTMLElement>('.coveo-category-facet-search-icon');
+      }
+
+      beforeEach(async done => {
+        facetElement = getFacetElement();
+        getFacetColumn().appendChild(facetElement.el);
+        await afterDeferredQuerySuccess();
+        done();
+      });
+
+      it('should have good contrast on the search button', () => {
+        const borderContrast = ContrastChecker.getContrastWithBackground(getSearchButton(), 'borderBottomColor');
+        expect(borderContrast).toBeGreaterThan(ContrastChecker.MinimumContrastRatio);
+      });
+
+      describe('after focusing on the search button', () => {
+        function hideFocusedSearchFacetValueStyle() {
+          const focusedClass = 'coveo-facet-search-current-result';
+          $$(getRoot())
+            .findClass(focusedClass)[0]
+            .classList.remove(focusedClass);
+        }
+
+        beforeEach(async done => {
+          (get(facetElement.el) as CategoryFacet).categoryFacetSearch.focus();
+          await waitUntilSelectorIsPresent(document.body, '.coveo-facet-search-results');
+          done();
+        });
+
+        it('search should be accessible', async done => {
+          hideFocusedSearchFacetValueStyle();
```

Comment:
This was added because axe was reporting a contrast issue on the text of a focused value (requiring a 4.5:1 contrast ratio). However, we already established that we aren't fixing contrast issues for focused suggestions.